mp4parse_capi/src/lib.rs

```diff
@@ -278,15 +278,15 @@ pub unsafe extern fn mp4parse_get_track_count(parser: *const mp4parse_parser, co
     MP4PARSE_OK
 }
 
-fn media_time_to_ms(time: MediaScaledTime, scale: MediaTimeScale) -> u64 {
+fn media_time_to_us(time: MediaScaledTime, scale: MediaTimeScale) -> Option<u64> {
     assert!(scale.0 != 0);
-    time.0 * 1000000 / scale.0
+    time.0.checked_mul(1000000).and_then(|t| t.checked_div(scale.0))
 }
 
-fn track_time_to_ms(time: TrackScaledTime, scale: TrackTimeScale) -> u64 {
+fn track_time_to_us(time: TrackScaledTime, scale: TrackTimeScale) -> Option<u64> {
     assert!(time.1 == scale.1);
     assert!(scale.0 != 0);
-    time.0 * 1000000 / scale.0
+    time.0.checked_mul(1000000).and_then(|t| t.checked_div(scale.0))
```

Comment:
I think we can do better than this if we take a similar approach to Chromium's TimeDeltaFromRational, e.g. https://crbug.com/345888 and the related change to track_run_iterator.cc.

I'm r-ed the libstagefright changes for the same reason.